in-cluster DNS and load balancers on more platforms

mhrivnak opened the pull request:
proposes adding in-cluster DNS and load balancers as an option for more platform types than just the current `baremetal` and `openstack`.

haircommander committed:
Add enhancement for supporting user namespaces

Signed-off-by: Peter Hunt <pehunt@redhat.com>

jakobmoellerdev committed:
fix: remove misleading .configurationPolicy=ExternalConfiguration

eranco74 commented on enhancements/network/in-cluster-dns-and-loadbalancers-on-more-platforms.md:
```diff
@@ -0,0 +1,297 @@
+  - @mhrivnak
+  - @eranco
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - TBD
```
- Ben Nemec
 - Igal Tsoiref
 - Zane Bitter

eranco74 commented on enhancements/network/in-cluster-dns-and-loadbalancers-on-more-platforms.md:
```diff
@@ -0,0 +1,297 @@
+metal, including a BareMetalHost resource for each Node, which can be confusing
+to users. Even the web console’s Overview landing page shows “Infrastructure
+provider: BareMetal” in addition to “N Nodes” and “N Bare Metal Hosts”.
+
```
Unsure if we want to add it here, but this enhancement should mention Node OpenShift deployment (although its a bit distracting).
In SNO  (non-platform), DNS is handled independently using dnsmasq (when deployed by assisted). This proposal can align the behavior when installing SNO by enabling DNS service (load balancer ins't required) independently of the platform type.

mhrivnak replied:
I assume "mention Node OpenShift" should say "mention Single Node OpenShift".

Is this what you're saying?

- since SNO uses platform type `none`, it typically requires the user to create the internal DNS records
- but assisted-installer alleviates that by configuring and enabling dnsmasq
- making dnsmasq enablement a native part of the `none` platform, as this proposal describes, would make that easier for all SNO users?

eranco74 replied:
Yes, that's great, need to find the right  place for it

eranco74 commented on enhancements/network/in-cluster-dns-and-loadbalancers-on-more-platforms.md:
```diff
@@ -0,0 +1,297 @@
+
+### Goals
+
+Enable multi-node OpenShift clusters to be viable out-of-the-box in use cases
```
Enable multi-node OpenShift clusters to be viable out-of-the-box in environments that lack suitable external DNS and/or load balancer solutions.

* I think we can drop  `and B) is not one of the platform types that already provides those
services in-cluster (baremetal, openstack, an ovirt).`

mhrivnak replied:
I like the wording simplification, dropping "use cases".

But I think it's worth highlighting that this has already been done for three platform types, the goal here is not to change anything about that, and that we're essentially proposing more of what's already been previously agreed as a goal.

eranco74 commented on enhancements/network/in-cluster-dns-and-loadbalancers-on-more-platforms.md:
```diff
@@ -0,0 +1,297 @@
+solution, and B) is not one of the platform types that already provides those
+services in-cluster (`baremetal`, `openstack`, an `ovirt`).
+
+Stop requiring users to select the `baremetal` platform type when all they
```
Allow users to opt-in for in-cluster DNS and load balancer services independently of platform type.

mhrivnak replied:
The proposal as-is doesn't do that. It's just adding the option to two additional platform types.

The alternative described at the bottom would make it independent of platform type, but I don't think that's actually a goal.

WDYT?

eranco74 replied:
I agree, I was just trying to rephrase the goal to be more positive.
Perhaps:
Allow users to opt-in for in-cluster DNS and load balancer services in None and External platform

openshift-merge-bot[bot] committed:
Merge pull request #1649 from haircommander/scc-userns

Add enhancement for supporting user namespaces

openshift-merge-bot[bot] committed:
Merge pull request #1663 from jakobmoellerdev/remove-lvms-microshift-external-config-flag

NOJIRA: fix: remove misleading .configurationPolicy=ExternalConfiguration

mhrivnak committed:
in-cluster DNS and load balancers on more platforms

proposes adding in-cluster DNS and load balancers as an option for more
platform types than just the current `baremetal` and `openstack`.
